asdf/_tests/core/_converters/test_complex.py

```diff
@@ -80,16 +80,18 @@ def test_valid_nan_complex(valid):
         pass
 
 
-def test_roundtrip():
+def test_roundtrip(tmp_path):
```

Comment:
The usage of `roundtrip_object` here does not play nice with the lazy tree. As the lazy tree isn't used here this is likely a left-over change from when I had it enabled by default. In this case, because `values` is the object round-tripped, `roundtrip_object` returns an `AsdfDictNode` for `values` and does not convert all of the contained complex numbers. Because `roundtrip_object` isn't a  context manager, the file is closed and when this test attempts to compare the contents of `values` an error will occur due to the closed file.